src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,165 @@
+            logger.warning(e)
+            if call_count != (num_attempts - 1):
+                time.sleep(cooloff)
+                cooloff *= 2
+                call_count += 1
+                continue
+            else:
+                raise
+
+        except requests.exceptions.HTTPError as e:
+            logger.warning(e)
+            if response.status_code == 404:
+                raise
+
+            logger.info(f"API return code {response.status_code} cooloff at {cooloff}")
+            if call_count != (num_attempts - 1):
+                time.sleep(cooloff)
+                cooloff *= 2
+                call_count += 1
+                continue
+            else:
+                raise
+
+        # We got through the loop without error so we've received a valid response
+        return response
+
+
+def request_with_cooloff(url, num_attempts=10):
+    return json.loads(_request_with_cooloff(url, num_attempts).content.decode("utf-8"))
+
+
+def get_yearly_mean_std(data_dict: dict, variable_dict: dict):
+    """
+    Get a df with yearly mean and std for every variable
+    """
+
+    df = pd.DataFrame(data_dict, index=pd.to_datetime(data_dict["time"]))
+    df.drop(columns="time", inplace=True)
+
+    for variable in variable_dict.keys():
+        variable_columns = [column for column in df.columns if variable in column]
+        df[variable] = df[variable_columns].mean(axis=1)
+        df.drop(columns=variable_columns, inplace=True)
+
+    return df.groupby(df.index.year).agg(["mean", "std"])
+
+
+def plot_data(df: pd.DataFrame, variable_dict: dict, city: str):
+    """
+    Plot variables for every city
+    """
+
+    fig, axes = plt.subplots(3, 1, figsize=(12, 8))
+
+    for axe, variable, unit in zip(axes, variable_dict.keys(), variable_dict.values()):
+        init_year = df.index[0]
+        end_year = df.index[-1]
+        axe.set_title(variable)
+        axe.set_ylabel(unit)
+
+        axe.plot(df[variable]["mean"], "-b", label="mean")
+        axe.plot(df[variable]["mean"] + df[variable]["std"], "--r", label="std")
+        axe.plot(df[variable]["mean"] - df[variable]["std"], "--r")
+        axe.set_xticks([])
+        axe.legend()
+
+    custom_xticks = [year for year in range(init_year, end_year, 5)]  # x label with two years spans
+
+    custom_xticks_labels = [str(year) for year in custom_xticks]  # x label with two years spans
+
+    axes[-1].set_xticks(custom_xticks)
+    axes[-1].set_xticklabels(labels=custom_xticks_labels)
+    plt.xticks(fontsize=10)
+    plt.xticks(rotation=45)
+
+    # Modify additional parameters for the entire figure
+    fig.suptitle(city, fontsize=16)
+    plt.savefig(f"src/module_1/{city}.png")
+    plt.close()
+
 
 def main():
-    raise NotImplementedError
+    start_date = "1950-01-01"
+    end_date = "2049-12-31"
+
+    cities = COORDINATES.keys()
+
+    for city in cities:
+        data_dict = get_data_meteo_api(city, start_date, end_date)
+
+        variable_dict = {}
```

Comment:
This is quite confusing while reading it with external eyes. I think it would help if `get_data_meteo_api` has a docstring showing the rough schema of the data coming from the API.

Also, this operation is very linked to `data_dict` and the API response itself. It may make sense to put everything into a function that first get the data from API and then processed it. If the response schema is sketched in docstring then it's easier to follow what the post-processing is doing